tartufo/scanner.py

```diff
@@ -476,3 +477,52 @@ def chunks(self):
         )
         for blob, file_path in self._iter_diff_index(diff_index):
             yield types.Chunk(blob, file_path)
+
+
+class FolderScanner(ScannerBase):
+    """Used to scan a folder."""
+
+    folder_options: types.FolderOptions
+    folder_path: str
+
+    def __init__(
+        self,
+        global_options: types.GlobalOptions,
+        folder_options: types.FolderOptions,
+        folder_path: str,
+    ) -> None:
+        """Used for scanning a folder.
+
+        :param global_options: The options provided to the top-level tartufo command
+        :param folder_options: The options specific to interacting with a folder
+        :param folder_path: The local filesystem path to scan
+        """
+        self.folder_options = folder_options
+        self.folder_path = folder_path
+        super().__init__(global_options)
+
+    @property
+    def chunks(self):
+        """Yield the individual file changes currently staged for commit.
+
+        :rtype: Generator[Chunk, None, None]
+        """
+
+        for blob, file_path in self._iter_folder():
+            if self.should_scan(file_path):
+                yield types.Chunk(blob, file_path)
+
+    def _iter_folder(self):
+        folder_path = pathlib.Path(self.folder_path)
+        for file_path in folder_path.glob(self.folder_options.pattern):
+            if file_path.is_file():
+                with open(file_path, "rb") as fhd:
```

Comment:
Neat trick with `Path` objects: You can actually use `with file_path.open("rb") as hfd:`. Not a blocker, just a handy usability thing.

Reply:
nice!